cmd/kube-controller-manager/app/discovery.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package app implements a server that runs a set of active
+// components.  This includes replication controllers, service endpoints and
+// nodes.
+//
+package app
+
+import (
+	"net/http"
+
+	endpointslicecontroller "k8s.io/kubernetes/pkg/controller/endpointslice"
+)
+
+func startEndpointSliceController(ctx ControllerContext) (http.Handler, bool, error) {
+	go endpointslicecontroller.NewController(
+		ctx.InformerFactory.Core().V1().Pods(),
+		ctx.InformerFactory.Core().V1().Services(),
+		ctx.InformerFactory.Core().V1().Nodes(),
+		ctx.InformerFactory.Discovery().V1alpha1().EndpointSlices(),
+		ctx.ComponentConfig.EndpointSliceController.MaxEndpointsPerSlice,
```

Comment:
don't we need to validate this?

Reply:
Good catch, I've added some basic validation logic for the controller config (https://github.com/robscott/kubernetes/blob/endpointslice-controller/cmd/kube-controller-manager/app/options/endpointslicecontroller.go#L68). Strangely it looks like the only other controller doing config validation is the StatefulSetController.